src/helper/worker_pool.cpp

```diff
@@ -3,40 +3,60 @@
 //
 // This product includes software developed at Datadog
 // (https://www.datadoghq.com/). Copyright 2021 Datadog, Inc.
+
 #include "worker_pool.hpp"
 
+using namespace std::chrono_literals;
+
 namespace dds::worker {
 
-void monitor::stop()
+namespace {
+
+void work_handler(consumer_queue &&q, std::optional<runnable> &&opt_r)
```

Comment:
`runnable` needn't be optional afaics

Reply:
No, but passing the optional allows me to rewrite the code below into something much more compact.